sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/IterableOfByteBuffersInputStream.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * An {@link InputStream} implementation that is based on {@link Iterator} of {@link java.nio.ByteBuffer}.
+ */
+public class IterableOfByteBuffersInputStream extends InputStream {
+
+    private final Iterator<ByteBuffer> buffers;
+    private ByteBuffer currentBuffer;
```

Comment:
Does this need to be volatile?

Reply:
No. As long as all access is under synchronized blocks.